mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.multi
+    def message_post(
+        self, body='', subject=None,
+        message_type='notification', subtype=None,
+        parent_id=False, attachments=None,
+        notif_layout=False, add_sign=True, model_description=False,
+        mail_auto_delete=True, **kwargs
+    ):
+        vals = super(MailThread, self).message_post(
+            body=body, subject=subject,
+            message_type=message_type, subtype=subtype,
+            parent_id=parent_id, attachments=attachments,
+            notif_layout=notif_layout, add_sign=add_sign, model_description=model_description,
+            mail_auto_delete=mail_auto_delete, **kwargs
+        )
+        mail_template = self.env["mail.message.custom.subject"].search([
+            ("model_id.model", "=", self._name),
```

Comment:
```suggestion
            ("model_id.model", "=", vals['model']),
```

Reply:
I ended up not applying this because, after the latest re-factor, I don't have `vals` available, and `self._name` seems like the best option.